src/pybind/mgr/call_home_agent/module.py

```diff
@@ -0,0 +1,322 @@
+    """
+    return report_inventory()
+
+
+class Report:
+    def __init__(self, name: str, fn: Callable[[], str], url: str, minutes_interval: int):
+        self.name = name                       # name of the report
+        self.fn = fn                           # function used to retrieve the data
+        self.url = url                         # url to send the report
+        self.interval = minutes_interval * 60  # interval to send the report (seconds)
+
+    def print(self) -> str:
+        if reports_header['private_key'] and reports_header['api_key']:
+            return self.fn()
+        else:
+            raise Exception('Not able to print <%s> report. Identification keys are not available' % self.name)
+
+    def send(self) -> str:
+        fail_reason = ""
+        resp = None
+        if reports_header['private_key'] and reports_header['api_key']:
+            try:
+                resp = requests.post(url=self.url,
+                                     headers={'accept': 'application/json', 'content-type': 'application/json'},
+                                     data=self.fn())
+                resp.raise_for_status()
+            except Exception as e:
+                explanation = resp.content if resp else ''
+                fail_reason = 'Failed to send <%s> to <%s>: %s \n%s' % (self.name, self.url, str(e), explanation)
+        else:
+            fail_reason = 'Not able to send <%s> report. Identification keys are not available' % self.name
+        return fail_reason
+
+
+class CallHomeAgent(MgrModule):
+    MODULE_OPTIONS: List[Option] = [
+        Option(
+            name='ches_endpoint',
+            type='str',
+            default=CHES_ENDPOINT,
+            desc='Call Home Event streamer end point'
+        ),
+        Option(
+            name='interval_inventory_report_minutes',
+            type='int',
+            default=INTERVAL_INVENTORY_REPORT_MINUTES,
+            desc='Time frequency for the inventory report'
+        ),
+        Option(
+            name='interval_performance_report_minutes',
+            type='int',
+            default=INTERVAL_PERFORMANCE_REPORT_MINUTES,
+            desc='Time frequency for the performance report'
+        ),
+        Option(
+            name='customer_email',
+            type='str',
+            default='',
+            desc='Customer contact email'
+        ),
+        Option(
+            name='country_code',
+            type='str',
+            default='',
+            desc='Customer country code'
+        )
+    ]
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super(CallHomeAgent, self).__init__(*args, **kwargs)
+
+        # set up some members to enable the serve() method and shutdown()
+        self.run = True
+
+        # Init module options
+        # Env vars (if they exist) have preference over module options
+        self.ches_url = str(os.environ.get('CHA_CHES_ENDPOINT', self.get_module_option('ches_endpoint')))
```

Comment:
Nothing to do with different config in different managers, this is just to simplify configuration in k8s envs.
Providing the config options as env vars coming from a k8s config map in the manager container is the most easy way to config the manager module in this kind of envs.

Reply:
The way this is implemented right now, it would always prefer the environment variables over the Ceph config. Maybe it would be better to read out the env vars at start, write them into Ceph config and then use whatever is in the Ceph config? I imagine this would make it easier for us later on, to dynamically change config values on the fly without restarting the module.